### DISCUSSION TITLE
Absolutely Best Model

### DISCUSSION THREAD

**Discussion by fakerybakery** [asigalov61/Euterpe-X#5] [Q&A] [answered]
Hi @asigalov61,
Would you mind providing a list of the absolutely highest quality music generation implementation? Your profile README says Allegro Music Transformer is the "best model/implementation overall," but it says this (Euterpe X) is the "most stable and capable model/implantation."
Might you be able to provide a list of the most capable implementation, and most efficient one?
Thanks!

**Comment by asigalov61** [comment, marked as answer]
@fakerybakery Sure, np! Let me know if you still have any quesitons...

Allegro Music Transformer is the best model OVERALL!!! This is means it plays well and it is the most efficient one at the same time.

Euterpe X (Large in particular) is the most capable and stable model but it is slower and less efficient than Allegro Music Transformer models.

So if you need high quality, use Euterpe X Large, if you need efficiency with good quality, use Allegro.

Hope this helps but feel free to ask questions :)

Alex.

**Reply by fakerybakery** [reply]
Ok, thanks so much!

**Comment by asigalov61** [comment]
@fakerybakery Hey,

Just wanted to let you know that I further increased the generation speed for Euterpe X. Check it out. I think you will enjoy it :)

Especially check out how it generates at max memory tokens/max seq len. It is blazing fast now, especially on A100.

Any feedback would be appreciated.

Alex.
